dj/models/node.py

```diff
@@ -245,6 +255,16 @@ class AvailabilityState(AvailabilityStateBase, table=True):  # type: ignore
         default_factory=partial(datetime.now, timezone.utc),
     )
 
+    def is_available(
+        self,
+        criteria: Optional[BuildCriteria] = None,  # pylint: disable=unused-argument
+    ) -> bool:  # pragma: no cover
+        """
+        Determine whether an availability state is useable given criteria
+        """
+        # Criteria to determine if an availability state should be used needs to be added
```

Comment:
You don't like the `# TODO: ...` marker?

Reply:
I do but we have a pre-commit check which will fail if it finds any to-do's. I know this is basically an attempt at circumventing that but I'm expecting to follow-up pretty quickly with an implementation of this function. (famous last words, I know...)